### ISSUE TITLE
model.summary() output shape is "multiple" when defining input_shape via model.build()

### ISSUE THREAD

**Issue by brett-daley** [keras-team/keras#13782]
When I define a model and pass the `input_shape` to the first layer, the Output Shape is well-defined after I call `model.summary()`. However, if I define a model and then pass the `input_shape` to `model.build()`, the Output Shape displays as "multiple." This behavior does not make sense to me. Both models should be identical as far as I can tell.

For example, running this code snippet

```python
import tensorflow as tf
from tensorflow.keras.layers import Dense

if __name__ == '__main__':
    model1 = tf.keras.Sequential([Dense(1, input_shape=[10])])
    print(model1.summary())

    model2 = tf.keras.Sequential([Dense(1)])
    model2.build(input_shape=[None, 10])
    print(model2.summary())
```

produces this output:

```
Model: "sequential"
_________________________________________________________________
Layer (type)                 Output Shape              Param #
=================================================================
dense (Dense)                (None, 1)                 11
=================================================================
Total params: 11
Trainable params: 11
Non-trainable params: 0
_________________________________________________________________
None
Model: "sequential_1"
_________________________________________________________________
Layer (type)                 Output Shape              Param #
=================================================================
dense_1 (Dense)              multiple                  11
=================================================================
Total params: 11
Trainable params: 11
Non-trainable params: 0
_________________________________________________________________
None
```

**System information**  
- Have I written custom code (as opposed to using example directory):  **yes**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  **Linux Ubuntu 18.04**
- TensorFlow backend (yes / no):  **yes**
- TensorFlow version:  **2.1.0**
- Keras version:  **2.3.1**
- Python version:  **3.6.9**
- CUDA/cuDNN version:  **CUDA 10.0 / cuDNN 7.6.5**
- GPU model and memory:  **NVIDIA GeForce RTX 2080 Ti (11 Gb)**

**Comment by theblackfly** [comment]
Why is it that `input_shape` does not include the batch dimension when passed as an argument to the `Dense` layer but DOES include  the batch dimension when `input_shape` is passed to the `build` method?

**Comment by admshumar** [comment]
Any movement on this issue? I'm refactoring a variational autoencoder to subclass from ks.Model and am having difficulties debugging it, partly because of this issue.

**Comment by BinyanHu** [comment]
Any update? Shape of the layes should be known as after `model.build` is called.

**Comment by theblackfly** [comment]
The problem seems to have been fixed. I get:
```
Model: "sequential"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
dense (Dense)                (None, 1)                 11        
=================================================================
Total params: 11
Trainable params: 11
Non-trainable params: 0
_________________________________________________________________
Model: "sequential_1"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
dense_1 (Dense)              (None, 1)                 11        
=================================================================
Total params: 11
Trainable params: 11
Non-trainable params: 0
_________________________________________________________________
```

* OS Platform and Distribution: **Linux Ubuntu 20.04.1 LTS x86_64**
* Python version: **3.8.3**
* Tensorflow version: **2.3.1**
* Keras version: **2.4.0**

**Comment by brett-daley** [comment]
Thanks everyone who replied to this issue. As pointed out by @theblackfly, the behavior has been fixed in later versions of tensorflow. Out of curiosity, I tested the releases between 2.1.0 (mine) and 2.3.1 (theblackfly's); it looks like the bug was first fixed in 2.3.0.

**Comment by jayagami** [comment]
still remains in tf2.5?

**Comment by yoni12345678** [comment]
I use TensorFlow 2.5 and still get this error when Keras's model API `class my_model(tf.Keras.Model):`

My code:

`class text_extracture(tf.keras.Model):#This class inherits from tensorflow.keras.Model
  
  #this function defines model architecture and hyperparameters
  def __init__(self,
               num_blocks = 4,
               num_filters = [16, 32, 64, num_known_chars], 
               kernal_sizes = [(3, 3), (3, 3), (3, 3), (3, 3)],
               #stride_vals = [(1, 1), (1, 1), (1, 1), (1, 1)],
               padding_vals = ['same', 'same', 'same', 'same'],
               pool_vals = [(3, 2), (2, 2), (2, 2), (2, 2)],
               batch_size = batch_size,
               **kwargs):
    
    super(text_extracture, self).__init__(**kwargs)
    self.num_blocks = num_blocks
    self.batch_size = batch_size
    #input is of size (batch size, image height, image width, 3)
    self.lambda_layer = tf.keras.layers.Lambda(lambda x: tf.cast(x, tf.float32) / 255.)
    #self.resize_layer = tf.keras.layers.experimental.preprocessing.Resizing(6144, 4096)
    self.resize_layer = tf.keras.layers.experimental.preprocessing.Resizing(3072, 2048)

    self.conv_1 = tf.keras.layers.Conv2D(filters = num_filters[0], kernel_size = kernal_sizes[0], padding = padding_vals[0])
    self.batch_norm_1 = tf.keras.layers.BatchNormalization()
    self.max_pool_1 = tf.keras.layers.MaxPool2D(pool_size=pool_vals[0])

    self.conv_2 = tf.keras.layers.Conv2D(filters = num_filters[1], kernel_size = kernal_sizes[1], padding = padding_vals[1])
    self.batch_norm_2 = tf.keras.layers.BatchNormalization()
    self.max_pool_2 = tf.keras.layers.MaxPool2D()

    self.conv_3 = tf.keras.layers.Conv2D(filters = num_filters[2], kernel_size = kernal_sizes[2], padding = padding_vals[2])
    self.batch_norm_3 = tf.keras.layers.BatchNormalization()
    self.max_pool_3 = tf.keras.layers.MaxPool2D()

    self.conv_4 = tf.keras.layers.Conv2D(filters = num_filters[3], kernel_size = kernal_sizes[3], padding = padding_vals[3])
    self.batch_norm_4 = tf.keras.layers.BatchNormalization()
    self.max_pool_4 = tf.keras.layers.MaxPool2D()

    self.last_layer = tf.keras.layers.Lambda(lambda x: get_actual_prediction(x))
  
  def call(self, inputs):
    x = tf.keras.layers.InputLayer(input_shape = (self.batch_size, None, None, 3))(inputs)
    x = self.lambda_layer(x)
    x = 1.0 - x
    x = self.resize_layer(x)

    x = self.conv_1(x)
    x = self.batch_norm_1(x)
    x = tf.keras.activations.relu(x)
    x = self.max_pool_1(x)

    x = self.conv_2(x)
    x = self.batch_norm_2(x)
    x = tf.keras.activations.relu(x)
    x = self.max_pool_2(x)

    x = self.conv_3(x)
    x = self.batch_norm_3(x)
    x = tf.keras.activations.relu(x)
    x = self.max_pool_3(x)

    x = self.conv_4(x)
    x = self.batch_norm_4(x)
    x = tf.keras.activations.relu(x)
    x = self.max_pool_4(x)
    
    #Do the rest for the rest of the layers
    x = tf.keras.layers.Flatten()(x)
    x = tf.keras.layers.Reshape(target_shape = (16384, num_known_chars))(x)
    x = tf.keras.activations.sigmoid(x)
    x = self.last_layer(x)
    return x

model = text_extracture()
model.build((5, 4096, 3048, 3))
model.summary()`

The output:

_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
lambda_58 (Lambda)           multiple                  0         
_________________________________________________________________
resizing_29 (Resizing)       multiple                  0         
_________________________________________________________________
conv2d_113 (Conv2D)          multiple                  448       
_________________________________________________________________
batch_normalization_41 (Batc multiple                  64        
_________________________________________________________________
max_pooling2d_112 (MaxPoolin multiple                  0         
_________________________________________________________________
conv2d_114 (Conv2D)          multiple                  4640      
_________________________________________________________________
batch_normalization_42 (Batc multiple                  128       
_________________________________________________________________
max_pooling2d_113 (MaxPoolin multiple                  0         
_________________________________________________________________
conv2d_115 (Conv2D)          multiple                  18496     
_________________________________________________________________
batch_normalization_43 (Batc multiple                  256       
_________________________________________________________________
max_pooling2d_114 (MaxPoolin multiple                  0         
_________________________________________________________________
conv2d_116 (Conv2D)          multiple                  64624     
_________________________________________________________________
batch_normalization_44 (Batc multiple                  448       
_________________________________________________________________
max_pooling2d_115 (MaxPoolin multiple                  0         
_________________________________________________________________
lambda_59 (Lambda)           multiple                  0

**Comment by arubiales** [comment]
Still in Tensorflow 2.6

**Comment by afrendeiro** [comment]
Hi, I'm still having this issue with Tensorflow 2.5.
Any chance to re-open the issue as multiple users report this is still ongoing?

**Comment by tarquo23** [comment]
Same issue on TensorFlow 2.6.0

**Comment by jinuhwang** [comment]
Same issue on TensorFlow 2.7.0

**Comment by kangsan0420** [comment]
Same issue on TensorFlow 2.8.0

**Comment by geroldcsendes** [comment]
Same issue on TensorFlow 2.4.1

**Comment by dhesenkamp** [comment]
Same issue on TensorFlow 2.8.0

**Comment by agniszczotka** [comment]
same issue Tensorflow 2.9.1 , cpu version, mac

**Comment by godhj93** [comment]
I'm using following code to make "model.summary" work.

```
class DeepNN(tf.keras.Model):
    
    def __init__(self):
        
        super(DeepNN, self).__init__()    
    
        self.flat = tf.keras.layers.Flatten()
        self.fc1 = tf.keras.layers.Dense(10,activation='relu')
        self.fc2 = tf.keras.layers.Dense(10,activation='relu')
        

    def call(self, x):
        
        x = self.flat(x)
        x = self.fc1(x)
        x = self.fc2(x)
        
        return x
    
    def make(self, input_shape=(32,32,3)):
        '''
        This method makes the command "model.summary()" work.
        input_shape: (H,W,C), do not specify batch B
        '''
        x = tf.keras.layers.Input(shape=input_shape)
        model = tf.keras.Model(inputs=[x], outputs=self.call(x), name='actor')
        print(model.summary())
        return 

```

**Comment by ehkarimi** [comment]
still in 2.10.0 tensorflow-cpu

**Comment by Ayazzia01** [comment]
Same issue 2.9.2 tf version on cpu, google colab pro

**Comment by widium** [comment]
With Tensorflow version 2.7.0 its work by Re-Initialize Model inside `__init__` method

```python
from keras.layers import Input, Dense
from keras.models import Model

class TestModel(Model):

    def __init__(self, nbr_dense, units, **kwargs):
        """Initialize Model instance with :
        - Parameters like : nbr_dense, units
        - Instance of Layer like : self.dense = Dense(..)
        """
        "Initialize Class with all method and attribute from Model Class"
        super(TestModel, self).__init__(**kwargs)
        
        "Define All Parameters"
        self.nbr_dense = nbr_dense
        self.units = units

        "Instance all Layer"
        self.input_layer = Input(shape=(7), name="input_layer")
        self.output_layer = Dense(units=1, activation="sigmoid", name=f"Output_Dense_Layer")

        
        ## ============ Construct Network ============= ##
		## Re-Initialize Model with input_layer inside call method to setup all shape
        input_shape = self.input_layer
        outputs_shape = self.call(input_shape)

        super(TestModel, self).__init__(inputs=input_shape,
                                        outputs=outputs_shape,
                                        **kwargs)
		## ============ ================== ============= ##
		
    def call(self, input):
        "Connect Layer Together and pass information"
        x = input
        
        output = self.output_layer(x) 

        return (output)
```
```python
test_model = TestModel(nbr_dense=10,
                                            units=32)

test_model.summary()
```

**Comment by MeghnaNatraj** [comment]
Same issue on TensorFlow 2.14.0. Can this issue be re-opened?

**Comment by ikram-md** [comment]
Same issue on `tensorflow 2.15.0`. This issue needs to be re-opened, unless the input shape is specified in the first layer, `model.build()` does not compute it automatically for the next layers.
